components/com_content/src/Model/CategoryFormModel.php

```diff
@@ -0,0 +1,501 @@
+            $this->state->set('category.id', $id);
+            $this->state->set('category.parent_id', $parentId);
+            $this->state->set('category.extension', $extension);
+            $this->state->set('category.component', $parts[0]);
+            $this->state->set('category.section', $parts[1] ?? null);
+            $this->state->set('params', $app->getParams());
+            $this->state->set('return_page', base64_decode($input->get('return', '', 'base64')));
+            $this->state->set('layout', $input->getCmd('layout', 'edit'));
+
+            if (!$id && Multilanguage::isEnabled()) {
+                $this->state->set('category.language', Factory::getLanguage()->getTag());
+            }
+
+            $this->__state_set = true;
+        }
+
+        return $property === null ? $this->state : $this->state->get($property, $default);
+    }
+
+    /**
+     * Method to auto-populate the model state.
+     *
+     * @return  void
+     *
+     * @since   6.1.0
+     */
+    protected function populateState()
+    {
+        // State is set in getState override to avoid recursion during debugging.
+    }
+
+    /**
+     * Method to get category data.
+     *
+     * @param   integer  $pk  The id of the category.
+     *
+     * @return  mixed  Category item data object on success, false on failure.
+     *
+     * @since   6.1.0
+     */
+    public function getItem($pk = null)
+    {
+        $pk = $pk ?: (int) ($this->getState($this->getName() . '.id') ?: $this->getState('category.id'));
+
+        $item = parent::getItem($pk);
+
+        if (!$item) {
+            return false;
+        }
+
+        $user         = $this->getCurrentUser();
+        $userId       = $user->id;
+        $asset        = $item->id ? 'com_content.category.' . $item->id : 'com_content';
+        $frontendEdit = Access::check($userId, 'core.edit.frontend', $asset);
+
+        $canEdit = false;
+
+        if ($frontendEdit !== false) {
+            if (!$item->id) {
+                $parentId = (int) ($item->parent_id ?? 0);
+                $canEdit  = $user->authorise('core.create', $asset)
+                    || $user->authorise('core.create', 'com_content.category.' . $parentId);
+            } elseif ($user->authorise('core.edit', $asset)) {
+                $canEdit = true;
+            } elseif (!empty($userId) && $user->authorise('core.edit.own', $asset)) {
+                $canEdit = (int) $userId === (int) $item->created_user_id;
+            }
+        }
+
+        $item->accessEdit   = $canEdit;
+        $item->accessChange = $frontendEdit !== false && $user->authorise('core.edit.state', $asset);
+
+        if ($item->id) {
+            $tagsHelper = new TagsHelper();
+            $item->tags = $tagsHelper->getTagIds($item->id, $item->extension . '.category');
+        }
+
+        return $item;
+    }
+
+    /**
+     * Get the return URL.
+     *
+     * @return  string  The return URL.
+     *
+     * @since   6.1.0
```

Comment:
```suggestion
     * @since   __DEPLOY_VERSION__
```

doing this means our build scripts will correctly set the version - it might not make it to 6.1 ;)